qulice-pmd/src/test/java/com/qulice/pmd/PMDEmptyTest.java

```diff
@@ -0,0 +1,218 @@
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty statement not in a loop fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStatementNotInLoopTest() throws Exception {
+        this.validatePMD(
+            "EmptyStatementNotInLoop.java",
+            Matchers.containsString(
+                "An empty statement (semicolon) not part of a loop"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty synchronized statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySynchronizedBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptySynchronizedBlock.java",
+            Matchers.containsString(
+                "Avoid empty synchronized blocks"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty switch statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySwitchStmtTest() throws Exception {
+        this.validatePMD(
+            "EmptySwitchStmt.java",
+            Matchers.containsString(
+                "Avoid empty switch statements"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty finally blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyFinallyBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptyFinallyBlock.java",
+            Matchers.containsString("Avoid empty finally blocks")
+        );
+    }
+
+    /**
+     * Makes sure that empty while statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyWhileStmtTest() throws Exception {
+        this.validatePMD(
+            "EmptyWhileStmt.java",
+            Matchers.containsString("Avoid empty while statements")
+        );
+    }
+
+    /**
+     * Makes sure that empty if blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyIfStmtTest() throws Exception {
+        this.validatePMD(
+            "EmptyIfStmt.java",
+            Matchers.containsString("Avoid empty if statements")
+        );
+    }
+
+    /**
+     * Makes sure that empty catch blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyCatchBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptyCatchBlock.java",
+            Matchers.containsString("Avoid empty catch blocks")
+        );
+    }
+
+    /**
+     * Validates that PMD reported given violation.
+     * @param file String containing file name to emulate.
+     * @param matcher Matching string needed in the output.
+     * @throws Exception If something wrong happens inside.
+     */
+    private void validatePMD(final String file,
```

Comment:
@prahladyeri This code is duplicated from `PMDValidatorTest`, let's avoid that (DRY principle). Better to place these tests in that file.